src/msg/async/frames_v2.h

```diff
@@ -837,6 +855,34 @@ struct MessageFrame : public Frame<MessageFrame,
   using Frame::Frame;
 };
 
+struct CompressionRequestFrame : public ControlFrame<CompressionRequestFrame,
+                                              bool, // is compress
+                                              std::vector<uint32_t>> { // preferred methods
+  static const Tag tag = Tag::COMPRESSION_REQUEST;
+  using ControlFrame::Encode;
+  using ControlFrame::Decode;
+
+  inline bool &is_compress() { return get_val<0>(); }
```

Comment:
what is the point of returning an reference?

Reply:
would be ideal if you could add more constness here.

Reply:
@tchaikov 
this is the convention that has been used in all other message frames.
I agree with you regarding your suggestions and questions - but I believe these changes should be discussed in a separate PR